common-libs/user-management/src/TrueCallerService.ts

```diff
@@ -65,7 +29,23 @@ export class TrueCallerService {
       throw new SdkErrorFromCode('UM-6')
     }
 
-    const keyResult = await this.fetchTrueCallerPublicKey(this.trueCallerUrl)
+    const res = await this.verifySignature(profileTrueCaller)
+
+    if (res) {
```

Comment:
```suggestion
    if (res) {
      return res
    }
    await this.keyManager.sync()
    return await this.verifySignature(profileTrueCaller)
```

Reply:
But it seems this is not needed at all as `await this.keyManager.getKey()` is performed in `verifySignature` already.